### PR TITLE
Add a new implementation of GeoTransforms based on cesium-native's LocalHorizontalCoordinateSystem

### PR DESCRIPTION
**Pull request by kring**
And tests to demonstrate it gets the same results as the old implementation. You can run the tests by going to Tools -> Test Automation in the Editor. Filter down to "Product Tests" and look for "Cesium".

So far this doesn't switch to the new implementation. I'll do a followup PR to use the new one, remove the old one, and remove the tests that verify they're consistent.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @kring!

* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] [Cesium for Unreal Samples](https://github.com/CesiumGS/cesium-unreal-samples) works.

File: Source/CesiumRuntime/Private/Tests/TestGeoTransforms.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+BEGIN_DEFINE_SPEC(
+    FTestGeoTransforms,
+    "Cesium.GeoTransforms",
+    EAutomationTestFlags::ApplicationContextMask |
+        EAutomationTestFlags::ProductFilter)
+END_DEFINE_SPEC(FTestGeoTransforms)
+
+void FTestGeoTransforms::Define() {
+  Describe("TransformLongitudeLatitudeHeightToUnreal", [this]() {
```
Comment (j9liu) [inline Source/CesiumRuntime/Private/Tests/TestGeoTransforms.cpp:18]: Might be nice to add more unit tests for `GeoTransformsNew`? Doesn't have to hold up this PR, it's already a step up from none 😛

**Review by j9liu — COMMENTED** [review]
@kring looks good to me! I just had a couple comments.

For future reference, I had to enable the "Functional Testing Editor" plugin in order to see the "Test Automation" option.

**Comment by j9liu** [comment]
@kring Since most of my suggestions were small, I added a commit for them. I'll wait for CI to pass before I merge this. In the meantime, feel free to comment or push back on any edits I made.

**Comment by kring** [comment]
Thanks @j9liu, your changes look good to me!
